tensorrt_llm/_torch/models/modeling_deepseekv3.py

```diff
@@ -457,25 +457,33 @@ def should_enable_alltoall(model_config: ModelConfig, top_k: int) -> bool:
     def compute_routed_output(self, hidden_states, hidden_states_fp4,
                               all_rank_num_tokens, cutlass_min_latency_mode):
         # max-throughput
+        use_dp_padding = False
         if self.use_dp and self.mapping.tp_size > 1:
-            max_num_token = max(all_rank_num_tokens)
-            hidden_states = torch.nn.functional.pad(
-                hidden_states,
-                (0, 0, 0, max_num_token - hidden_states.shape[0]))
             # FP4 all_gather moves this bf16 allgather in to after topk and fp4 quantization
             # to reduce allreduce BW
             if disable_fp4_allgather() and not self.enable_alltoall:
-                hidden_states = allgather(hidden_states,
-                                          self.mapping,
-                                          gather_dim=0)
+                hidden_states = allgather(
+                    hidden_states,
+                    self.mapping,
+                    gather_dim=0,
+                    all_rank_split_size=all_rank_num_tokens)
+            elif not self.experts.is_cutlass() or (not self.experts.has_fp8_qdq
+                                                   and self.experts.has_nvfp4):
+                # Use padding when not using the cutlass path or when x_sf in self.experts is not None
```

Comment:
why cutlass path requires padding?
While the otehr one doesn't? It's tokenwise op right?